File: shared/schemas/commit.py
# shared/schemas/commit.py
from datetime import datetime
from typing import List, Optional

from pydantic import BaseModel, ConfigDict, Field

from .enums import CommitIngestionStatusEnum, FileChangeTypeEnum
from .inference_job import InferenceJobRead


class CommitFileDiffRead(BaseModel):
    id: int
    file_path: str
    change_type: FileChangeTypeEnum
    old_path: Optional[str] = None
    diff_text: str

    model_config = ConfigDict(from_attributes=True, use_enum_values=True)


class CommitDetailsRead(BaseModel):
    id: int
    commit_hash: str
    author_name: str
    author_email: str
    author_date: datetime
    message: str
    parents: List[str]
    stats_insertions: int
    stats_deletions: int
    stats_files_changed: int
    file_diffs: List[CommitFileDiffRead]

    model_config = ConfigDict(from_attributes=True, use_enum_values=True)


class CommitPageResponse(BaseModel):
    ingestion_status: CommitIngestionStatusEnum
    details: Optional[CommitDetailsRead] = None
    inference_jobs: List[InferenceJobRead] = Field(default_factory=list)
    celery_ingestion_task_id: Optional[str] = None


class CommitListItem(BaseModel):
    commit_hash: str
    author_name: str
    author_date: datetime
    message_short: str
    ingestion_status: CommitIngestionStatusEnum

    model_config = ConfigDict(from_attributes=True, use_enum_values=True)


class PaginatedCommitList(BaseModel):
    items: List[CommitListItem]
    total: int
    skip: Optional[int] = None
    limit: Optional[int] = None

File: frontend/types/api/commit.ts
// frontend/types/api/commit.ts
import { CommitIngestionStatusEnum, FileChangeTypeEnum } from "./enums";
import { InferenceJobRead } from "./inference-job";

export interface CommitFileDiffRead {
  id: number;
  file_path: string;
  change_type: FileChangeTypeEnum;
  old_path?: string | null;
  diff_text: string;
}

export interface CommitDetailsRead {
  id: number;
  commit_hash: string;
  author_name: string;
  author_email: string;
  author_date: string; // ISO String
  message: string;
  parents: string[];
  stats_insertions: number;
  stats_deletions: number;
  stats_files_changed: number;
  file_diffs: CommitFileDiffRead[];
}

export interface CommitPageResponse {
  ingestion_status: CommitIngestionStatusEnum;
  details?: CommitDetailsRead | null;
  inference_jobs?: InferenceJobRead[] | null;
  celery_ingestion_task_id?: string | null;
}

export interface CommitListItem {
  commit_hash: string;
  author_name: string;
  author_date: string; // ISO String
  message_short: string;
  ingestion_status: CommitIngestionStatusEnum;
}

export interface PaginatedCommitList {
  items: CommitListItem[];
  total: number;
  skip?: number;
  limit?: number;
}